File: db/src/main/java/org/apple/db/BaseEntity.java
package org.apple.db;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;

import java.time.LocalDateTime;

@MappedSuperclass
public abstract class BaseEntity {

    @Id
    @GeneratedValue
    private Long id;

    private LocalDateTime createAt;

    private LocalDateTime updateAt;
}

File: db/src/main/kotlin/org/apple/db/BaseEntity.kt
package org.apple.db

import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.MappedSuperclass
import java.time.LocalDateTime

@MappedSuperclass
abstract class BaseEntity(
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    val createdAt: LocalDateTime? = null,

    var updatedAt: LocalDateTime? = null,
)